exonum/src/events/tests.rs

```diff
@@ -247,12 +247,14 @@ fn test_network_handshake() {
     let mut connect_list = ConnectList::default();
 
     let mut t1 = ConnectionParams::from_address(first);
-    connect_list.add(t1.connect_info);
+    connect_list.add(t1.connect_info.clone());
 
     let mut t2 = ConnectionParams::from_address(second);
-    connect_list.add(t2.connect_info);
+    connect_list.add(t2.connect_info.clone());
 
-    let connect_list = SharedConnectList::from_connect_list(connect_list);
+    let mut connect_list = SharedConnectList::from_connect_list(connect_list);
+    connect_list.resolve_and_cache_peer_address(&first.to_string());
+    connect_list.resolve_and_cache_peer_address(&second.to_string());
```

Comment:
Addresses are resolved by node in the connect method. Here we have connect list test in isolation so we have this calls here.
There is only one place where address is resolved -> in node handler connect. You can't forget to resolve it, since input to those function is string address, if it fails to resolve then connect method fails.